### PR TITLE
Add tests for presentational table attributes

### LINKED ISSUE #11235: No tests for presentational table attributes
In https://crbug.com/703053 I looked for tests for `<table background>` but couldn't find any. https://software.hixie.ch/utilities/js/live-dom-viewer/saved/5974 is an ad-hoc test.

In https://html.spec.whatwg.org/multipage/rendering.html#tables-2, at least the `background`, `bgcolor` and `bordercolor` attributes on `table` and related elements appear to be untested. They could easily be tested using reftests. For example:

Test:
```html
...
<table background=/images/green.png>
...
```

Ref:
```html
...
<table style="background-image:url(/images/green.png)">
...
```

### PR DESCRIPTION
**Pull request by xiuqijix**
Add following tests (Fix #11235):

 - {table, thead, body, tfoot, tr, td, th}#background
 - {table, thead, body, tfoot, tr, td, th}#bgcolor
 - table#bordercolor
 - table#cellspacing
 - table#cellpadding
 - {table, tr, td, th}#height#{pixel, percentage}
 - col#width#{pixel, percentage}
 - {thead, body, tfoot, tr, td, th}#align#{center, middle, left, right, justify}
 - th#default text-align property is center

@foolip, PTAL.

### CODE REVIEW

**Comment by foolip** [comment]
This is a pretty big PR, would take me a while to review properly. Were these tests generated by some script for all the variants, so I could review that script instead?

**Comment by kaixinjxq** [comment]
Unfortunately, I didn't use a script to generate them, as they are somehow a bit different.

**Review by foolip — COMMENTED** [review]
I've run the tests in Chrome Dev using `./wpt run --binary `which google-chrome-unstable` chrome html/rendering/non-replaced-elements/tables/` on my machine to see that they are passing.

However, running the tests I see that some of them don't fit in the 600x600 viewport, which makes it harder to have confidence that they are valid, as the important bits could possibly be outside of what's visible.

Trying to come up with a fix, I think that there's actually a better way to test these presentational attributes with a single testharness.js test.

First, have markup that include at least one each of the table, thead, tbody, tfoot, tr, td, and th elements. Then set the content attributes, and use getComputedStyle to check that it has the effect described in the spec.

**Comment by foolip** [comment]
Note that I'll be gone until July 5, so a further lack of response is just because of that.

**Comment by foolip** [comment]
Ah, so what I had in mind, or do now, is something a bit more uniform across the tests. Basically, put this markup in every test:
```html
<table>
  <thead></thead>
  <tbody>
    <tr>
      <th></th>
      <td></td>
    </tr>
  </tbody>
  <tfoot></tfoot>
</table>
```

In other words, one of each element. Then, each test has as its input an element name, an attribute name, a CSS property, and an expected value. The test would set the attribute, call `getComputedStyle(elem).getPropertyValue(cssProp)` and compare that to the expected value.

If you add a cleanup step in each test to remove the attribute, you could put all of these tests in a single file, with the test inputs in a big array.

**Comment by kaixinjxq** [comment]
Yeah, thanks @foolip, that's a good idea! Lots of tests based on same `table` element,  they can be merged together. But there are differences in some specific checkpoints,  such as:
- The `table` element `cellspacing` attribute maps to property `border-spacing` on the` table` element, and the `table` element `cellpadding` attribute maps to porperty `padding` on `td and th` elements,  the affected object are different between `cellspacing` and `cellpadding`.
- Only `table/tr/td/th` have the `height` attribute, and only `thead/body/tfoot/tr/td/th` have the `align` attribute.
- When we test the `width` attribute of the `col` element, we need use the `col` element, but most table attribute tests don't.
- And so on.

If we move all tests input in one array, it would be more complex.
So I just move all the tests in one test file to use one common markup and merge the uniform parts into several tests, PTAL.

**Comment by kaixinjxq** [comment]
@foolip, friendly ping.

**Review by foolip — COMMENTED** [review]
Sorry for glacial speed of review. Looks good, just don't think we should test for prefixes.

File: html/rendering/non-replaced-elements/tables/table-attribute.html [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+</div>
+
+<script>
+
+const ids = ["table", "thead", "tbody", "tfoot", "tr", "td", "th"];
+const alignIds = ["thead", "tbody", "tfoot", "tr", "td", "th"];
+const heightIds = ["tr", "td", "th"];
+const div = document.getElementById("div");
+const table = document.getElementById("table");
+const prefixes = ["-moz-", "-ms-", "-o-", "-webkit-", ""];
```
[the remark this answers is not in the document]
Comment (kaixinjxq) [reply html/rendering/non-replaced-elements/tables/table-attribute.html]: No relevant prefixed description was found in spec. I just submit a new commit to remove it.
Unfortunately, most tests failed without these prefixes.

**Review by foolip — APPROVED** [review]

**Comment by foolip** [comment]
@kaixinjxq finally merged, thanks for your patience!

**Pull request merged**